### PR TITLE
Update CI configs to v0.4.17

### PR DESCRIPTION
**Pull request by pionbot**
Update lint scripts and CI configs.
Excluding codecov.yml.

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -25,6 +35,7 @@ jobs:
         run: |
           go test \
           -coverprofile=cover.out -covermode=atomic \
+          -bench=. \
```
Comment (at-wat) [inline .github/workflows/test.yaml:38]: Benchmark test timed out.

Comment (Sean-Der) [reply .github/workflows/test.yaml:38]: I have just been ignoring adding `-bench` to this repo.

I want to get a consistent performance story across Pion repos, but just haven't had the time. I was thinking maybe we could use `benchstat`? https://github.com/swaggest/rest/blob/master/.github/workflows/bench.yml or https://github.com/ipfs/ci-helpers/blob/master/orbs/ci-go/commands/benchtest.yml

If performance increases (or decreases) 10% it requires updating the test results in tree?

Comment (at-wat) [reply .github/workflows/test.yaml:38]: I think the setting which is not shareable among all repos shouldn't be in .goassets.
I had added .ci.conf file for each repo to control repository specific setting, but it seems ignored now.

Comment (Sean-Der) [reply .github/workflows/test.yaml:38]: 100% agree, the github actions migration just missed that. All the bash files still respect them though! https://github.com/pion/.goassets/issues/44

**Comment by Sean-Der** [comment]
v0.4.18 has been released

**Pull request closed without merging**